src/overlays/actors/ovl_Boss_Mo/z_boss_mo.c

```diff
@@ -22,299 +128,2943 @@ const ActorInit Boss_Mo_InitVars = {
+                gSPSegment(POLY_XLU_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(gDust1Tex));
+                gDPSetEnvColor(POLY_XLU_DISP++, 250, 250, 255, 0);
+                gSPDisplayList(POLY_XLU_DISP++, gMorphaDropletSetupDL);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Mo/func_809206C4.s")
+                flag++;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Mo/func_80921280.s")
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, (s16)(*effect).vShimmer, (s16)(*effect).vShimmer, 0xFF,
+                            effect->alpha);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Mo/func_809216D0.s")
+            Matrix_Translate(effect->pos.x, effect->pos.y, effect->pos.z, MTXMODE_NEW);
+            Matrix_Scale(effect->scale, 1.0f, effect->scale, MTXMODE_APPLY);
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(gfxCtx, "../z_boss_mo.c", 7441),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Mo/func_80922D30.s")
+            gSPDisplayList(POLY_XLU_DISP++, gMorphaWetSpotVtxDL);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Mo/BossMo_Update.s")
+    effect = effectHead;
+    flag = 0;
+    for (i = 0; i < 300; i++, effect++) {
+        if (effect->type == MO_FX_BUBBLE) {
+            if (flag == 0) {
+                func_80093D18(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Mo/func_809237C4.s")
+                gDPSetEnvColor(POLY_OPA_DISP++, 150, 150, 150, 0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Mo/func_80923870.s")
+                flag++;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Mo/func_80923FDC.s")
+            gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, effect->alpha);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Mo/func_80924228.s")
+            Matrix_Translate(effect->pos.x, effect->pos.y, effect->pos.z, MTXMODE_NEW);
+            func_800D1FD4(&globalCtx->mf_11DA0);
+            Matrix_Scale(effect->scale, effect->scale, 1.0f, MTXMODE_APPLY);
+            gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(gfxCtx, "../z_boss_mo.c", 7476),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Mo/BossMo_Draw.s")
+            gSPDisplayList(POLY_OPA_DISP++, gMorphaBubbleDL);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Mo/func_80924D70.s")
+    Matrix_Pop();
+    CLOSE_DISPS(gfxCtx, "../z_boss_mo.c", 7482);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Mo/func_80925480.s")
+void BossMo_Unknown(void) {
+    // Called every frame by UpdateCore, but nothing sets the BREGs.
+    // Testing in PJ64 indicates this function effectively does nothing.
```

Comment:
Did you test it while actually setting the BREGs from the reg editor? I would expect this function to be used for sfx testing during development, and we should probably name and/or comment it as such.